### PR TITLE
GFW v2 MK 1

### PR DESCRIPTION
**Pull request by vgarcia13**
JIRA Ticket: https://allenai.atlassian.net/browse/GUNDI-3061

**NOTES:**

- I left all the logic for calling GFW endpoints and methods (auth_header, api_keys, aoi, geostore, etc). I just slightly changed some instructions there for a better reading
- All the logic for the geometries support was left untouched as well
- The "chunking" process may not be working as expected, I am still on that

**NOTE:** Tests with integrated alerts are still in process.

### CODE REVIEW

File: app/actions/handlers.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                    state = {
+                        "latest_device_timestamp": vehicle.get("recorded_at")
+                    }
+                    await state_manager.set_state(
+                        str(integration.id),
+                        "pull_observations",
+                        state,
+                        vehicle.get("source")
+                    )
+                """
```
Comment (chrisdoehring) [inline app/actions/handlers.py:61]: Let's remove this.

Comment (vgarcia13) [reply app/actions/handlers.py:61]: But we are going to save this:

```
{
  "gfw_integrated_alerts": {
    "dataset": "gfw_integrated_alerts",
    "latest_updated_on": "2024-05-08T15:33:47.485916+00:00",
    "version": "v20240211"
  },
  "latest_cartodb_id": 0
}
```

for the integrated alerts right? That's why I left that commented

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,189 @@
+#
+# This file is autogenerated by pip-compile with Python 3.11
```
Comment (marianobrc) [inline requirements.txt:2]: The Integrations template works with Python 3.8, and that's the version in the [Dockerfile](https://github.com/PADAS/gundi-integration-action-runner/blob/602756b7902a86b505c8d94f6d6797b46387a023/docker/Dockerfile#L1). As we are using some features of python 3.10 here, maybe we can upgrade to python 3.10 in the template repo.
cc @chrisdoehring

File: app/actions/client.py [hunk 3]
```diff
@@ -0,0 +1,725 @@
...
+                "attention_needed": True
+            }
+        )
+        raise e
+    else:
+        logger.info(f"Got {len(alerts)} fire alerts")
+        return alerts
+
+
+async def get_glad_alerts(aoi_data, integration, config):
```
Comment (chrisdoehring) [inline app/actions/client.py:616]: These are no longer GLAD alert, but are instead "Integrated Alerts". 
Perhaps we can take this opportunity to update the names in code.

File: app/actions/client.py [hunk 4]
```diff
@@ -0,0 +1,725 @@
...
+
+from datetime import datetime, timedelta, timezone
+from typing import Optional, List, Set, Tuple
+
+logger = logging.getLogger(__name__)
+state_manager = IntegrationStateManager()
+
+
+DEFAULT_FIRE_LOOKBACK_DAYS = 1
+DEFAULT_GLAD_LOOKBACK_DAYS = 1
```
Comment (chrisdoehring) [inline app/actions/client.py:26]: This should be a higher number, probably 10 days.

We might want to have these two `_LOOKBACK_DAYS` be configurable, with some sensible maximums.

Comment (vgarcia13) [reply app/actions/client.py:26]: We can have that variable to apply to both types? or we should ask for both in the config?

File: requirements.txt (hunk 2 above)
Comment (chrisdoehring) [reply requirements.txt:2]: Thanks @marianobrc Good idea.

File: app/actions/handlers.py [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+                            config=action_config
+                        )
+                    )
+                    integrated_alerts = await integrated_alerts_task
+                    if integrated_alerts:
+                        transformed_data = []
+                        logger.info(f"Integrated alerts pulled with success.")
+                        n = 0
+                        while n < len(integrated_alerts):
+                            transformed_integrated_alert = await transform_integrated_alert(integrated_alerts[n:n+4])
```
Comment (chrisdoehring) [inline app/actions/handlers.py:241]: @vgarcia13 this is not quite what I had in mind and I think we need to remove this. Apologies if my earlier comments were misleading.

I would like to simply create these Events individually as point events, just as we were doing before.

Later, in a separate story, we will come back and add a feature to create Geometry Events. I'll explain how to do that later.

**Review by chrisdoehring — APPROVED** [review]

**Pull request merged**